lib/Endpoint.js

```diff
@@ -97,9 +97,16 @@ class Endpoint extends EventEmitter {
       frame = ZCLMfgSpecificHeader.fromBuffer(rawFrame);
     } else frame = ZCLStandardHeader.fromBuffer(rawFrame);
 
-    const response = (frame.frameControl.disableDefaultResponse || (meta && meta.groupId))
-      ? null
-      : this._makeErrorResponse(frame);
+    // NOTE: we do not respond with a default response if:
+    // 1. The frame we received is a default response
+    // 2. Another command is sent in response to the received frame
+    // 3. The frame has the disableDefaultResponse flag set
+    // See ZCL specification 2.5.12.2.
+    const response = (
+      frame.frameControl.disableDefaultResponse
+      || (meta && meta.groupId)
+      || frame.cmdId === 11
+    ) ? null : this._makeErrorResponse(frame);
```

Comment:
`11` is a default response frame? Maybe add that to the comment `1. The frame we received is a default response (cmdId 11)`

I tend to go both ways on this but this logic is quite dense, you could maybe make a function? on the other hand maybe that is over-abstraction...

```js
_getDefaultResponse(frame, meta) {
  // frame has disableDefaultResponse flag set
  if (frame.frameControl.disableDefaultResponse) {
    return null;
  }
  // Another command is sent in response to the received frame
  if (meta && meta.groupId) {
    return null;
  }
  // frame is a default response
  if (frame.cmdId === 11) {
    return null;
  }
  // make default response
  return this._makeErrorResponse(frame);
} 
```

Reply:
Thanks for the feedback, I added the `cmdId=11` reference in the comment, makes it clearer indeed. Hm right, I agree a lot of the logic is quite dense and we could/should improve on that. For now I will leave it as is, but maybe someday I'll make time to do some cleaning up.